File: lab-prog-web/src/main/java/com/example/labprog/onepage/entity/Usuario.java
package com.example.labprog.onepage.entity;

import jakarta.persistence.*;

@Entity
@Table(name = "usuarios")
public class Usuario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, nullable = false)
    private String login;

    @Column(nullable = false)
    private String senha;

    @Column(nullable = false)
    private String nome;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private TipoUsuario tipo;

    @Column(nullable = false)
    private Boolean aprovado = false; // Para cadastros pendentes

    public Usuario() {}

    public Usuario(String login, String senha, String nome, TipoUsuario tipo) {
        this.login = login;
        this.senha = senha;
        this.nome = nome;
        this.tipo = tipo;
        this.aprovado = tipo == TipoUsuario.MASTER; // Master sempre aprovado
    }

    // Getters e Setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getLogin() { return login; }
    public void setLogin(String login) { this.login = login; }

    public String getSenha() { return senha; }
    public void setSenha(String senha) { this.senha = senha; }

    public String getNome() { return nome; }
    public void setNome(String nome) { this.nome = nome; }

    public TipoUsuario getTipo() { return tipo; }
    public void setTipo(TipoUsuario tipo) { this.tipo = tipo; }

    public Boolean getAprovado() { return aprovado; }
    public void setAprovado(Boolean aprovado) { this.aprovado = aprovado; }
}

File: app grifo att/lib/models/usuario.dart
class Usuario {
  final int id;
  final String login;
  final String nome;
  final String tipo;
  final bool aprovado;

  Usuario({
    required this.id,
    required this.login,
    required this.nome,
    required this.tipo,
    required this.aprovado,
  });

  factory Usuario.fromJson(Map<String, dynamic> json) {
    return Usuario(
      id: json['id'] ?? 0,
      login: json['login'] ?? '',
      nome: json['nome'] ?? '',
      tipo: json['tipo'] ?? 'MEMBRO',
      aprovado: json['aprovado'] ?? false,
    );
  }

  String get tipoFormatado {
    switch (tipo) {
      case 'MASTER':
        return 'Master';
      case 'ADMIN':
        return 'Administrador';
      case 'MEMBRO':
        return 'Membro';
      default:
        return tipo;
    }
  }
}

